gloo/test/base_test.h

```diff
@@ -105,10 +121,16 @@ class BaseTest : public ::testing::Test {
       int base = 2) {
     Barrier barrier(size);
     ::gloo::rendezvous::HashStore store;
+
+    auto device = createDevice(transport);
+    if (!device) {
+      return;
```

Comment:
I'd like to -- but there are some tests that only work with the TCP transport and are still included in the build. Otherwise we have to make compilation of those conditional on the availability of the TCP transport (which will be unavailable on macOS and Windows).